### ISSUE TITLE
[BUG]: streamdeck_ui.lock is never released/deleted

### ISSUE THREAD

**Issue by zen2** [streamdeck-linux-gui/streamdeck-linux-gui#154]
### In raising this issue I confirm that

- [X] I have fully completed the issue template
- [X] I have searched open and closed issues for duplicates
- [X] I have read the [Contribution Guidelines](https://streamdeck-linux-gui.github.io/streamdeck-linux-gui/contributing/contributing-guide/)
- [X] I have read the [Code of Conduct](https://streamdeck-linux-gui.github.io/streamdeck-linux-gui/contributing/code-of-conduct/)
- [X] I have read the [Documentation](https://streamdeck-linux-gui.github.io/streamdeck-linux-gui/)

### Describe the bug

Once streamdeck-ui have exited properly by using "exit" in interface or sending signal SIGTERM to terminate the process you still can find the lock file streamdeck_ui.lock (/tmp/streamdeck_ui.lock on my system).

### Steps to reproduce the behavior

1.    Launch streamdeck-ui
2.    Exit streamdeck-ui
3.    ls -l /tmp/streamdeck_ui.lock

### Expected behavior

 Lock file shouldn't exist any more once streamdeck-ui have terminated.

### System Information

*    OS: Gentoo Linux
*    Desktop environment: openbox
*    Python version: 3.11
*    streamdeckui version: 2.0.15

### Stream Deck Version

XL

**Comment by ajbucci** [comment]
I would chalk this up as a design decision, not a bug. The application uses a semaphore. The presence of the file is not what provides the locking mechanism; it's the acquisition of the lock via fcntl.flock on the file descriptor.

**Comment by zen2** [comment]
Even if it a semaphore the file used for locking need to be deleted on application exit.
/tmp is not mandatory flushed especially if it is not a tmpfs mount point.

**Comment by ajbucci** [comment]
I would agree that while deleting the lock file on application exit is generally good practice for cleanliness and clarity, in this specific scenario, given the use of the /tmp directory and the advisory nature of the locking mechanism, not deleting the file should not pose a significant issue. 

Other than the file persisting in your /tmp directory are you experiencing any adverse impacts? For example, the application being unable to acquire a lock when it otherwise should be able to.

**Comment by cstromblad** [comment]
I decided to try and contribute something with a small pull request for this "event". https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/pull/209

It certainly has no real effect on the application, but it does appear better to remove the file once the application terminates.

**Comment by tdcam** [comment]
I have multiple accounts on my Linux workstation, one for work, and one for personal use. I log in for work in the morning, then I log out and log in with my personal account. Every day, I have to manually delete the lock and socket files before I can use my Streamdeck. It's pretty frustrating. 

Can we at maybe create the lock and socket files with maybe the username, like /tmp/workacct_streamdeck_ui.lock plus /tmp/workacct_streamdeck_ui.sock and /tmp/homeacct_streamdeck_ui.lock plus /tmp/homeacct_streamdeck_ui.sock? Then, even if you don't delete the lock and socket files on exit, it won't hurt anything.
